pkg/operator/ceph/operator.go

```diff
@@ -161,11 +163,26 @@ func (o *Operator) Run() error {
 	}
 }
 
-func (o *Operator) startSystemDaemons(clusterSpec *cephv1.ClusterSpec) error {
+func (o *Operator) startDrivers() error {
 	if o.delayedDaemonsStarted {
 		return nil
 	}
 
+	o.delayedDaemonsStarted = true
```

Comment:
Why not set this to true at the end of the method? Then you don't need to set it to false if it fails

Reply:
If you are updating already running CSI  driver on the midway, especially for CSI node plugin.
Need to test below scenarios
* how updating for daemonset pods behave if any PVC's are mounted to app pods. 
* Need to test things when fuse or ndb client are used for PVC  mounting
* Also you need to check the behavior because of `OnDelete` update strategy

Reply:
@Madhu-1 This would be the same scenario as any upgrade, right? In our docs for upgrades or for updating the csi settings in the configmap, we should be clear about the impact of the restart. 
@umangachapagain How about adding a link to [this section](https://rook.io/docs/rook/master/ceph-upgrade.html#2-csi-upgrade-pre-requisites) of the upgrade guide when documentation is added for the configmap?

Reply:
Sounds good.
I'll also try to improve Integration Test to catch anything odd.